PWGHF/TableProducer/candidateCreatorSigmac0plusplusCascade.cxx

```diff
@@ -0,0 +1,345 @@
+#include "Common/Core/TrackSelection.h"
+#include "Common/Core/trackUtilities.h"
+#include "Common/Core/TrackSelectionDefaults.h"
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::analysis;
+using namespace o2::constants::physics;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+struct HfCandidateCreatorSigmac0plusplusCascade {
+  Configurable<float> trkMinPt{"trkMinPt", 0.15, "track min pT"};
+  Configurable<float> trkMaxEta{"trkMaxEta", 0.8, "track max Eta"};
+  Configurable<float> maxDCAxyToPVcut{"maxDCAxyToPVcut", 2.0, "Track DCAxy cut to PV Maximum"};
+  Configurable<float> maxDCAzToPVcut{"maxDCAzToPVcut", 2.0, "Track DCAz cut to PV Maximum"};
+  Configurable<float> nTpcNClsFound{"nTpcNClsFound", 120, "nFindable TPC Clusters"};
+  Configurable<float> nTPCCrossedRows{"nTPCCrossedRows", 70, "nCrossed TPC Rows"};
+  Configurable<float> nTPCChi2{"nTPCChi2", 4.0, "nTPC Chi2 per Cluster"};
+  Configurable<float> nITSChi2{"nITSChi2", 36.0, "nITS Chi2 per Cluster"};
+  Configurable<float> tpcnSigmaPi{"tpcnSigmaPi", 3.0, "TPC nSigma selection"};
+
+  /// Selection of candidates Λc+
+  Configurable<int> selectionFlagLc{"selectionFlagLc", 1, "Selection Flag for Lc"};
+  Configurable<double> yCandLcMax{"yCandLcMax", -1., "max. candLc. Lc rapidity"};
+  Configurable<std::vector<double>> binsPt{"binsPt", std::vector<double>{hf_cuts_sigmac_to_p_k_pi::vecBinsPt}, "pT bin limits"};
+  Configurable<int> selectionFlagLcToK0sP{"selectionFlagLcToK0sP", 1, "Selection Flag for Lc"};
+  Configurable<int> selectionFlagLcbarToK0sP{"selectionFlagLcbarToK0sP", 1, "Selection Flag for Lcbar"};
+  Configurable<double> cutsMassLcMax{"cutsMassLcMax", 0.08, "Lc candidate mass selection"};
+
+  /// Selections on candidate soft π-,+
+  Configurable<bool> applyGlobalTrkWoDcaCutsSoftPi{"applyGlobalTrkWoDcaCutsSoftPi", false, "Switch on the application of the global-track w/o dca cuts for soft pion BEFORE ALL OTHER CUSTOM CUTS"};
+  Configurable<float> softPiEtaMax{"softPiEtaMax", 0.9f, "Soft pion max value for pseudorapidity (abs vale)"};
+  Configurable<float> softPiChi2Max{"softPiChi2Max", 36.f, "Soft pion max value for chi2 ITS"};
+  Configurable<int> softPiItsHitMap{"softPiItsHitMap", 127, "Soft pion ITS hitmap"};
+  Configurable<int> softPiItsHitsMin{"softPiItsHitsMin", 1, "Minimum number of ITS layers crossed by the soft pion among those in \"softPiItsHitMap\""};
+  Configurable<float> softPiDcaXYMax{"softPiDcaXYMax", 0.065, "Soft pion max dcaXY (cm)"};
+  Configurable<float> softPiDcaZMax{"softPiDcaZMax", 0.065, "Soft pion max dcaZ (cm)"};
+  Configurable<bool> addQA{"addQA", true, "Switch for the qa PLOTS"};
+
+  /// Table with Σc0,++ info
+  Produces<aod::HfCandScCasBase> rowScCandidates;
+
+  template <typename TrackType>
+  bool isTrackSelected(const TrackType& track)
+  {
```

Comment:
Do not defined methods in between data member declarations or before `init`. See the PWGHF guidelines.

Reply:
Done.